lib/segment/src/segment/entry.rs

```diff
@@ -430,17 +430,7 @@ impl SegmentEntry for Segment {
         self.segment_type
     }
 
-    fn info(&self) -> SegmentInfo {
-        let payload_index = self.payload_index.borrow();
-        let schema = payload_index
-            .indexed_fields()
-            .into_iter()
-            .map(|(key, index_schema)| {
-                let points_count = payload_index.indexed_points(&key);
-                (key, PayloadIndexInfo::new(index_schema, points_count))
-            })
-            .collect();
-
+    fn size_info(&self) -> SegmentInfo {
```

Comment:
We might want a separate type here to prevent accidental misuse?

Reply:
Since we're not differentiating between `info` and `size_info()` in proxy segments and this type being used directly in the API, I tried to avoid changing this type and thought a proper comment would suffice.

If you prefer having a separate type, I'll implement it and try to not break the API (which could be possible using `#[serde(flatten)]` but haven't checked)

Reply:
> If you prefer having a separate type, I'll implement it and try to not break the API (which could be possible using #[serde(flatten)] but haven't checked)

I'd also be fine with some overlap of fields between two types, that may be easier without flattening.

But I don't have a strong preference on having a separate type or not. I'd also be happy to keep it for now.